### ISSUE TITLE
Cannot tune sequential model given get different results for each run

### ISSUE THREAD

**Issue by fickas** [tensorflow/tensorflow#60768]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

binary

### Tensorflow Version

2.12

### Custom Code

No

### Current Behaviour?

I am trying to tune a sequential model. But if I rerun my tuning code, I get a different result for "best parameters" each time. I am not using GPUs. I have seeds set to a constant.

My tuning code ran fine under an earlier tf version. But when I updated to latest, it stopped producing replicable results.

Did a specific version give up on replicability? Is there a workaround for tuning?

### Standalone code to reproduce the issue

```shell
https://colab.research.google.com/drive/1wxa5PCf7kQhTKrDFlbehq-k8MGAQpwkK?usp=sharing
```

### Relevant log output

_No response_</details>

**Comment by medmabcf** [comment]
It is common for the results of a machine learning model to vary slightly between runs, even when setting random seeds to a constant. While setting random seeds can help control for some sources of randomness and make your results more consistent between runs, it is still possible for the results to vary slightly. This is normal and to be expected. also the weights of a machine learning model will not be exactly the same after each run. The weights are initialized randomly at the beginning of training and are updated during training based on the data and the optimization algorithm used. As a result, the final weights of the model can vary slightly between runs.
I can see that the results you got were not changing significantly. 
Hopefully this was helpful

**Comment by fickas** [comment]
Still a bit concerned about tuning. When I ran simple tuning algorithm on a Sequential model, I came out with different architectures as best on every rerun, i.e., different number of layers and different nodes in each layer. It is kind of hard to average architectures across reruns.

My conjecture is this will also affect sklearn tuning algorithms, e.g., RandomSearch and HalvingSearch. They will not produce consistent "best parameters" on rerun. I plan to test this out in next couple days. I feel I am missing something because I would expect a hue and cry from the community if tuning libraries stopped working as expected and I have not seen any complaints.

Also, it still seems something changed in a version update. As I said, I would get consistent results with early versions of 2, but an update to latest 2 version changed things.

**Comment by tilakrayal** [comment]
@fickas,
Thank you for opening this issue. Development of keras moved to another [repository](https://github.com/keras-team/keras/issues). 


Could you please post this issue on keras-team/keras [repo](https://github.com/keras-team/keras/issues).
To know more please refer:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!

**Comment by tilakrayal** [comment]
Calling [tf.keras.utils.set_random_seed](https://www.tensorflow.org/api_docs/python/tf/keras/utils/set_random_seed) sets the Python seed, the NumPy seed, and the TensorFlow seed. Setting these seeds is necessary to ensure any random numbers your program generates are also deterministic.

I tried to execute the code with the alternative approach by using tf.keras.utils.set_random_seed, and I was getting the same result while executing the mentioned code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/d906cd28355958b4ea371cee7ba2866a/different_results.ipynb).

Could you please have a look at this official document for reference.
https://www.tensorflow.org/api_docs/python/tf/config/experimental/enable_op_determinism

Thank you!
